.circleci/config.yml

```diff
@@ -0,0 +1,171 @@
+
+  build:
+    environment:
+      - TZ: "/usr/share/zoneinfo/America/Los_Angeles"
+    docker:
+      - image: docker:18.01.0-ce-git
+    working_directory: /tmp/src
+    steps:
+      - run:
+          name: Install parallel gzip and python3
+          command: apk add --no-cache pigz python3
+      - restore_cache:
+          keys:
+            - docker-v1-{{ .Branch }}
+          paths:
+            - /tmp/cache/container.tar.gz
+      - restore_cache:
+          key: dependency-cache-repo2docker
+      - checkout
+      - setup_remote_docker
+      - run:
+          name: Load Docker image layer cache
+          no_output_timeout: 30m
+          command: |
+            docker info
+            set +o pipefail
+            if [ -f /tmp/cache/container.tar.gz ]; then
+              pigz -d --stdout /tmp/cache/container.tar.gz | docker load
+            fi
+            docker images
+      - run:
+          name: Run jupyter-repo2docker to generate Docker Container
+          no_output_timeout: 60m
+          command: |
+            echo "0. Installing dependencies..."
+            python3 -m venv venv
+            . venv/bin/activate
+            pip install jupyter-repo2docker
+            export VERSION=$(python3 -c "import repo2docker; print(repo2docker.__version__)")
+            echo "repo2docker version: ${VERSION}"
+            echo "1. Preparing building environment"
+            echo "Working directory is ${PWD}"
+            # If not set, define CONTAINER_NAME
+            if [ ! -n "${CONTAINER_NAME:-}" ]
+                then
+                    CONTAINER_NAME="${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}"
+            fi
+            # If not set, use testing repo name
+            if [ ! -n "{REPO_NAME}" ]
+              then
+              ${REPO_NAME}="https://www.github.com/${CONTAINER_NAME}"
+            fi
+            echo "Container name is ${CONTAINER_NAME}"
+            echo "Repository name is ${REPO_NAME}"
+            echo "2. Running jupyter-repo2docker..."
+            echo "jupyter-repo2docker --debug --user-name username --user-id 1000 --no-run --image-name ${CONTAINER_NAME} ${REPO_NAME}"
+            jupyter-repo2docker --debug --user-name username --user-id 1000 --no-run --image-name "${CONTAINER_NAME}" "${REPO_NAME}"
+            echo "3. Running without server (example)"
+            docker run -td --name repo2docker --entrypoint /bin/bash "${CONTAINER_NAME}"
+            docker ps
+            docker images
+      - run:
+          name: Docker Save
+          no_output_timeout: 40m
+          command: |
+            # If not set, define CONTAINER_NAME
+            if [ ! -n "${CONTAINER_NAME:-}" ]
+                then
+                    CONTAINER_NAME="${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}"
+            fi
+            echo "Saving ${CONTAINER_NAME} to container.tar.gz"
+            mkdir -p /tmp/cache
+            docker save ${CONTAINER_NAME}:latest \
```

Comment:
Will this save the state after running the command on L77 or state before? Both could be interesting to have but I'd add a comment to clarify which is going to be save here.

Reply:
What is L77?

Reply:
I know what this is now :)

Reply:
I thought you were talking about some derivation of Fortran 77 (F77 -> L77) and I just had no idea.